modules/service/src/main/resources/db/migration/V0379__execution_event_view.sql

```diff
@@ -0,0 +1,80 @@
+CREATE TYPE e_execution_event_type AS ENUM(
+  'sequence',
+  'step',
+  'dataset'
+);
+
+-- Combine the sequence, step and dataset event tables into one view.
+-- We do this to make GraphQL queries that return all events easy to implement.
+CREATE VIEW v_execution_event AS
```

Comment:
Perhaps I should just make `t_execution_event` with these columns instead of creating this view?  Each table, `t_sequence_event`, `t_step_event` and `t_dataset_event` has a mutually exclusive `c_execution_event_id` column.

Reply:
Sharing the same id sequence is kind of a smell I guess, but it could really go either way. The combined table wouldn't be absurdly wide (11 columns, I think?) but the check constraint will be tedious to write and maintain, and the Scala end may be a little more complex. I don't think the union will be significantly slower, but that's just a guess. We could always change it later, without affecting the GraphQL schema.

So, 🤷🏻 ... I don't have a good answer.